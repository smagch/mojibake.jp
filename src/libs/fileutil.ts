export function cutFileExtension(filename: string): string {
  const pos = filename.lastIndexOf(".");
  if (pos === -1) {
    return "";
  }
  return filename.slice(pos);
}
