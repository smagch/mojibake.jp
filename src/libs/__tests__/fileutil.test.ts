import { cutFileExtension } from "../fileutil";

test.each`
  filename
  ${"foo/bar/hoge.txt"}
  ${"foo.csv"}
  ${"foo/bar/hoge"}
  ${"bar"}
  ${""}
`('cutFileExtension("$filename")', ({ filename }) => {
  const ext = cutFileExtension(filename);
  expect(ext).toMatchSnapshot();
});
