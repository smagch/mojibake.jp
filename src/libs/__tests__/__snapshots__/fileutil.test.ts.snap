// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cutFileExtension("") 1`] = `""`;

exports[`cutFileExtension("bar") 1`] = `""`;

exports[`cutFileExtension("foo.csv") 1`] = `".csv"`;

exports[`cutFileExtension("foo/bar/hoge") 1`] = `""`;

exports[`cutFileExtension("foo/bar/hoge.txt") 1`] = `".txt"`;
